src/Framework/FieldsAPI/FormField/FieldOptions.php

```diff
@@ -2,29 +2,90 @@
 
 namespace Give\Framework\FieldsAPI\FormField;
 
-use Give\Framework\FieldsAPI\FormField\FieldTypes;
+use Give\Helpers\Arr;
 
 trait FieldOptions {
 
-	/** @var array */
+	/** @var FieldOption[] */
 	protected $options = [];
 
+	/**
+	 * Does the field type support options?
+	 *
+	 * @return bool
+	 */
 	public function supportsOptions() {
 		return in_array(
 			$this->getType(),
 			[
 				FieldTypes::TYPE_SELECT,
 				FieldTypes::TYPE_RADIO,
-			]
+			],
+			true
 		);
 	}
 
-	public function options( $options ) {
-		$this->options = $options;
+	/**
+	 * Set the options
+	 *
+	 * @param array $options
+	 *
+	 * @return $this
+	 */
+	public function options( array $options ) {
+		// Reset options, since they are meant to be set immutably
+		$this->options = [];
+
+		// Determine if the provided options are an associative array before the loop to avoid extra work.
+		$areOptionsAnAssocArray = Arr::isAssoc( $options );
+
+		// Loop through the options and transform them to the proper format.
+		foreach ( $options as $key => $value ) {
+			if ( $value instanceof FieldOption ) {
+				// In this case, what is provided matches the proper format, so we can just append it.
+				$this->options[] = $value;
+			} elseif ( is_array( $value ) ) {
+				// In this case, what has been provided in the value is an array with a value then a label.
+				// This matches the constructor of `FieldOption`, so we can unpack it as arguments for a new instance.
+				$this->options[] = new FieldOption( ...$value );
+			} elseif ( $areOptionsAnAssocArray ) {
+				// In this case, the array is associative, therefore, we expect the key to be the option’s value
+				// and the value to be the option’s label.
+				$this->options[] = new FieldOption( $key, $value );
```

Comment:
@knowler Let's just drop this. I'd rather not have too many options for... options. The class method is obvious, the list method is quick and clear, and the tuple method is robust. The associate method doesn't add anything and has limitations.

Reply:
Sounds good. I agree that though associative arrays _seem_ like a natural fit for options, their limitations create a number of issues, and ultimately they prove to be not a good fit.

Reply:
Resolved by 99ce87f